tests/unit/test_exit_if_null.c

```diff
@@ -95,10 +83,14 @@ test_exit_if_null_ok(void **state)
 {
 	char  *p;
 
-	// Trick: vasprintf(3) will actually return the new string, not 0.
-	will_return(__wrap_vasprintf, 0);
+	static const char  foo[] = "foo1bar";
 
-	p = xaprintf("foo%d%s", 1, "bar");
+	p = XMALLOC(countof(foo), char);
+	assert_true(p != NULL);
+	strcpy(p, foo);
 	assert_string_equal(p, "foo1bar");
 	free(p);
+
+	p = XMALLOC(0, char);
```

Comment:
Allocating 0 bytes has implementation-defined behavior (may return NULL or a valid pointer). This test case may be unreliable across different platforms

Reply:
That sounds really good. I'd like this to have a defined behaviour everywhere